raco/rules.py

```diff
@@ -15,8 +17,24 @@ class Rule(object):
 
     __metaclass__ = ABCMeta
 
+    _disabled_rules = set()
+    _flag_pattern = re.compile(r'no_([A-Za-z_]+)')  # e.g., no_MergeSelects
+
     def __call__(self, expr):
-        return self.fire(expr)
+        if not hasattr(expr, '_disabled'):
```

Comment:
I'm really not a fan of all this magic. Can you at least write a little comment that explains what is happening and why?